prime-router/metadata/fhir_mapping/hl7/datatypes/XPN/HumanName.yml

```diff
@@ -0,0 +1,208 @@
+  expressionsMap:
+    url:
+      type: SYSTEM_URL
+      value: humanname-assembly-order
+    valueCode:
+      type: STRING
+      valueOf: XPN.11
+      expressionType: HL7Spec
+
+# XPN.10 is deprecated so the schema favors using XPN.12/13 if either are populated
+period_2:
+  vars:
+    xpn12: String, XPN.12
+    xpn13: String, XPN.13
+  condition: $xpn12 NOT_NULL || $xpn13 NOT_NULL
+  expressionType: nested
+  expressionsMap:
+    start:
+      valueOf: "GeneralUtils.dateTimeWithZoneId(periodStart,ZONEID)"
+      expressionType: JEXL
+      vars:
+        periodStart: XPN.12
+    end:
+      valueOf: "GeneralUtils.dateTimeWithZoneId(periodEnd,ZONEID)"
+      expressionType: JEXL
+      vars:
+        periodEnd: XPN.13
+
+extension:
+  vars:
+    xpn2: String, XPN.2
+    xpn3: String, XPN.3
+    xpn4: String, XPN.4
+    xpn6: String, XPN.6
+    xpn7: String, XPN.7
+    xpn8: String, XPN.8
+    xpn9: String, XPN.9
+    xpn10: String, XPN.10
+    xpn12: String, XPN.12
+    xpn13: String, XPN.13
+    xpn14: String, XPN.14
+  condition: $xpn2 NOT_NULL || $xpn3 NOT_NULL || $xpn4 NOT_NULL || $xpn6 NOT_NULL || $xpn7 NOT_NULL || $xpn8 NOT_NULL || $xpn9 NOT_NULL || $xpn10 NOT_NULL || $xpn12 NOT_NULL || $xpn13 NOT_NULL || $xpn14 NOT_NULL
+  generateList: true
+  expressionType: nested
+  expressions:
+    - expressionType: nested
+      expressionsMap:
+        url:
+          type: SYSTEM_URL
+          valueOf: xpn2-given-name
+        valueString:
+          type: STRING
+          expressionType: HL7Spec
+          valueOf: XPN.2
+    - expressionType: nested
+      expressionsMap:
+        url:
+          type: SYSTEM_URL
+          valueOf: xpn3-second-given-name
+        valueString:
+          type: STRING
+          expressionType: HL7Spec
+          valueOf: XPN.3
+    - expressionType: nested
+      expressionsMap:
+        url:
+          type: SYSTEM_URL
+          valueOf: xpn4-suffix
+        valueString:
+          type: STRING
+          expressionType: HL7Spec
+          valueOf: XPN.4
+    - expressionType: nested
+      expressionsMap:
+        url:
+          type: SYSTEM_URL
+          valueOf: xpn6-degree
+        valueString:
+          type: STRING
+          expressionType: HL7Spec
+          valueOf: XPN.6
+    - expressionType: nested
+      expressionsMap:
+        url:
+          type: SYSTEM_URL
+          valueOf: xpn7-name-type-code
+        valueString:
+          type: STRING
+          expressionType: HL7Spec
+          valueOf: XPN.7
+    - expressionType: nested
+      expressionsMap:
+        url:
+          type: SYSTEM_URL
+          valueOf: xpn8-name-representation-code
+        valueString:
+          type: STRING
+          expressionType: HL7Spec
+          valueOf: XPN.8
+    - expressionType: nested
+      expressionsMap:
+        url:
+          type: SYSTEM_URL
+          valueOf: xpn9-name-concept
+        value:
+          type: STRING
+          expressionType: HL7Spec
+          valueOf: $xpn9
```

Comment:
I tested several methods including mapping to CWE[CodeableConcept] but was unable to retrieve values stored in `XPN.9`. There may be a limitation in the upstream libraries preventing full use of this field.

Reply:
Without a mapping, it seems fine to map it as a string, but can we add an integration test to demonstrate that this works? And if it doesn't work for CWEs generally, it might be better to just leave this out.

Reply:
Took another stab at it and got it working as CWE - I've added a test for it to one of the test files.